src/server/transaction.cc

```diff
@@ -1149,33 +1165,8 @@ pair<bool, bool> Transaction::ScheduleInShard(EngineShard* shard) {
   TxQueue::Iterator it = txq->Insert(this);
   DCHECK_EQ(TxQueue::kEnd, sd.pq_pos);
   sd.pq_pos = it;
-  unsigned q_limit = absl::GetFlag(FLAGS_tx_queue_warning_len);
 
-  if (txq->size() > q_limit) {
-    static thread_local time_t last_log_time = 0;
-    // TODO: glog provides LOG_EVERY_T, which uses precise clock.
-    // We should introduce inside helio LOG_PERIOD_ATLEAST macro that takes seconds and
-    // uses low precision clock.
-    time_t now = time(nullptr);
-    if (now >= last_log_time + 10) {
-      last_log_time = now;
-      EngineShard::TxQueueInfo info = shard->AnalyzeTxQueue();
-      string msg =
-          StrCat("TxQueue is too long. Tx count:", info.tx_total, ", armed:", info.tx_armed,
-                 ", runnable:", info.tx_runnable, ", total locks: ", info.total_locks,
-                 ", contended locks: ", info.contended_locks, "\n");
-      absl::StrAppend(&msg, "max contention score: ", info.max_contention_score,
-                      ", lock: ", info.max_contention_lock_name,
-                      ", poll_executions:", shard->stats().poll_execution_total);
-      const Transaction* cont_tx = shard->GetContTx();
-      if (cont_tx) {
-        absl::StrAppend(&msg, " continuation_tx: ", cont_tx->DebugId(), " ",
-                        cont_tx->IsArmedInShard(sid) ? " armed" : "");
-      }
-
-      LOG(WARNING) << msg;
-    }
-  }
+  AnalyzeTxQueue(shard, txq);
```

Comment:
nit: consider leaving `txq->size() > q_limit` check to avoid a redundant function call

Reply:
Oh no, it's much more readable. Single use function from anonymous namespace might be inlined, besides if we talk about this level of optimization there are much more of them, like reducing the number of atomics 🙄